File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.java
class Solution {
    public int[] solution(int[] numbers, String direction) {
        int len = numbers.length;
        int[] answer = new int[len];
        
        for(int i=0;i<len;i++){
            if(direction.equals("right")){
                if(i == 0) answer[0] = numbers[len-1];
                else answer[i] = numbers[i-1];
            }
            else{
                answer[i] = numbers[(i+1) % len];
            }
        }      
        return answer;
    }
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int numbers[], size_t numbers_len, const char* direction) {
    int* answer = (int*)malloc(numbers_len * sizeof(int));
    int i = 0;
    
    for(i=0;i<numbers_len;i++){
        if(strcmp(direction, "right") == 0){
            if(i==0) answer[0] = numbers[numbers_len-1];
            else answer[i] = numbers[i-1];
        }else{
            answer[i] = numbers[(i+1) % numbers_len];
        }
    }
    
    answer[i] = '\0';
    return answer;
}